core/blockchain.go

```diff
@@ -1940,8 +1947,19 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool) (int, error)
 			pstart = time.Now()
 			if bc.vmConfig.TxDAG == nil && bc.vmConfig.EnableParallelUnorderedMerge {
 				receipts, logs, usedGas, err = bc.serialProcessor.Process(block, statedb, bc.vmConfig)
+				blockProcessedInParallel = false
 			} else {
 				receipts, logs, usedGas, err = bc.processor.Process(block, statedb, bc.vmConfig)
+				blockProcessedInParallel = true
+				if err != nil {
+					// parallel processing fail , fallback to serial with new statDB.
+					log.Warn("ParallelEVM fallback!!!", "error", err.Error())
+					statedb, err = bc.reGenerateStateForFallBack(parent.Root, block.Root(), statedb)
```

Comment:
The error should be handled here because if err is not nil, then statedb will be nil. In this case, the next line `statedb.StartPrefetcher` could potentially cause a panic.

Reply:
Done